src/main/java/teammates/storage/sqlentity/User.java

```diff
@@ -57,10 +60,11 @@ protected User() {
         // required by Hibernate
     }
 
-    public User(Course course, Team team, String name, String email) {
+    public User(Course course, String name, String email) {
         this.setId(UUID.randomUUID());
+        this.courseId = course.getId();
```

Comment:
this should be in the setter of course

Reply:
In that case, we can leave Line 37 as it is?

Reply:
yes leave as is. idea is we should not be able to set course and courseId independently of each other.

Reply:
I understand, thanks for clarifying!!